test/bricks-net-api.test.js

```diff
@@ -0,0 +1,634 @@
+			
+			assert.equal(true, response instanceof api.HTTPResponse, 'response instanceof HTTPResponse');
+			assert.equal(true, when.isPromiseLike(response.body), 'response.body isPromiseLike');
+			
+			when(
+				response.body
+			).then(function (body) {
+				assert.equal('string', typeof body);
+				assert.equal('OK', body);
+				done();
+			}).done(undefined, done);
+		});
+		
+		it('should handle response Content-Length', function (done) {
+			serverRequestHandler = function (request, response) {
+				response.writeHead(200, {
+					'Content-Length': 5
+				});
+				response.write('OK');
+				response.write('OK2');
+				response.end('Overflow');
+			};
+			
+			var response = api.HTTP(api.GET("localhost:20000/test"));
+			
+			when(
+				response.body
+			).then(function (body) {
+				assert.equal('string', typeof body);
+				assert.equal('OKOK2', body);
+				done();
+			}).done(undefined, done);
+		});
+		
+		/* TODO(sompylasar): Investigate further on this test failure.
+		it('should throw ConnectionResetByPeer on early close', function (done) {
+			serverRequestHandler = function (request, response) {
+				response.writeHead(200, {
+					'Content-Length': 1000
+				});
+				response.end('Partial');
+				console.log('TEST ended.');
+			};
+			
+			var response = api.HTTP(api.GET("localhost:20000/test"));
+			
+			when(
+				response
+			).then(function (body) {
+				done(new Error('Resolved instead of rejected.'));
+			}, function (err) {
+				assert.equal(true, err instanceof ConnectionResetByPeer);
+				done();
+			}).done(undefined, done);
+		});
+		// */
+		
+		it('should make a `POST` request', function (done) {
+			serverRequestHandler = function (request, response) {
+				try {
+					assert.equal('POST', request.method);
+					assert.equal('/test', request.url);
+					assert.equal('1.1', request.httpVersion);
+					
+					// WARNING: `request.rawHeaders` requires Node.js 0.12.0
+					assert.deepEqual([
+						'Host',
+						'localhost'
+					], request.rawHeaders);
+					
+					response.end('OK');
+				}
+				catch (ex) {
+					done(ex);
+				}
+			};
+			
+			var response = api.HTTP(api.POST("localhost:20000/test"));
+			
+			assert.equal(true, response instanceof api.HTTPResponse, 'response instanceof HTTPResponse');
+			assert.equal(true, when.isPromiseLike(response.body), 'response.body isPromiseLike');
+			
+			when(
+				response.body
+			).then(function (body) {
+				assert.equal('string', typeof body, '`body` is a string');
+				assert.equal('OK', body, '`body` equals to the response body');
+				done();
+			}).done(undefined, done);
+		});
+		
+		it('should make a `POST` request with a body and a content type', function (done) {
+			var requestContentType = 'application/json';
+			var requestBody = '{"request":"body"}';
+			
+			serverRequestHandler = function (request, response) {
+				try {
+					assert.equal('POST', request.method);
+					assert.equal('/test', request.url);
+					assert.equal('1.1', request.httpVersion);
```

Comment:
Before I forget: `httpVersion` is not exposed by Bricks; should probably not show this field in REPL shell either.